internal/controller/build/controller.go

```diff
@@ -207,15 +215,55 @@ func (r *Reconciler) makeBuildContext(ctx context.Context, build *choreov1.Build
 	}, nil
 }
 
+func (r *Reconciler) getDPClient(ctx context.Context, build *choreov1.Build) (*client.Client, error) {
+	// Retrieve the dataplane associated with the environment
+	dataplaneRes, err := getDataplane(ctx, r.Client, build)
+	if err != nil {
+		// Return an error if dataplane retrieval fails
+		return nil, fmt.Errorf("failed to get default dataplane %s: %w", "default-dataplane", err)
+	}
+
+	// Get the DP client using the credentials from the dataplane
+	dpClient, err := dpKubernetes.GetDPClient(r.DpClientMgr, dataplaneRes)
+	if err != nil {
+		// Return an error if client creation fails
+		return nil, fmt.Errorf("failed to get DP client: %w", err)
+	}
+
+	// Return the DP client
+	return dpClient, nil
+}
+
+func getDataplane(ctx context.Context, c client.Client, build *choreov1.Build) (*choreov1.DataPlane, error) {
+	dataplaneList := &choreov1.DataPlaneList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(build.GetNamespace()),
+		client.MatchingLabels{
+			labels.LabelKeyOrganizationName: controller.GetOrganizationName(build),
+			labels.LabelKeyName:             "default-dataplane",
+		},
+	}
+
+	if err := c.List(ctx, dataplaneList, listOpts...); err != nil {
+		return nil, fmt.Errorf("failed to list dataplanes: %w", err)
+	}
+
+	if len(dataplaneList.Items) > 0 {
+		return &dataplaneList.Items[0], nil
+	}
+
+	return nil, fmt.Errorf("failed to find dataplanes")
+}
+
 // makeExternalResourceHandlers creates the chain of external resource handlers that are used to
 // create the build namespace and other resources required for argo workflows.
-func (r *Reconciler) makeExternalResourceHandlers() []dataplane.ResourceHandler[integrations.BuildContext] {
+func (r *Reconciler) makeExternalResourceHandlers(dpClient *client.Client) []dataplane.ResourceHandler[integrations.BuildContext] {
```

Comment:
```suggestion
func (r *Reconciler) makeExternalResourceHandlers(dpClient client.Client) []dataplane.ResourceHandler[integrations.BuildContext] {
```
since we always get the value of this, we don't need a pointer here.